proposals/local-warning-pragmas.rst

```diff
@@ -0,0 +1,178 @@
+Options Local Pragma
+====================
+
+.. proposal-number:: 
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/234>`_.
+.. sectnum::
+.. contents::
+
+We propose to add functionality for control warnings locally, in particular, add pragmas ``WARN``, ``IGNORE`` and ``ERROR``. For instance, in this code
+::
+ {-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+ f a =
+   {-# WARN name-shadowing #-}
+   \a -> a + a
+
+ g a = \a -> a + a
+
+there is ``-Wname-shadowing`` warning in funcion ``f`` but not in function ``g``.
+
+This proposal rises some ticket problems:
+ 1. https://gitlab.haskell.org/ghc/ghc/issues/602
+ 2. https://gitlab.haskell.org/ghc/ghc/issues/10150
+
+Motivation
```

Comment:
Incorporate that ton of motivating examples that others posted in the discussion.